src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -187,14 +191,25 @@ private static Rune ChangeCaseCultureAware(Rune rune, TextInfo textInfo, bool to
             original = original.Slice(0, charCount);
             modified = modified.Slice(0, charCount);
 
+#if CORECLR
             if (toUpper)
             {
-                textInfo.ChangeCaseToUpper(original, modified);
+                culture!.TextInfo.ChangeCaseToUpper(original, modified);
             }
             else
             {
-                textInfo.ChangeCaseToLower(original, modified);
+                culture!.TextInfo.ChangeCaseToLower(original, modified);
             }
```

Comment:
I'm on my phone and haven't looked at all call sites... is this potentially more expensive than it was previously, if callers were able to use a cached TextInfo?

Reply:
The place it was previously cached is in the `ToLowerInvariant/ToUpperInvariant` methods.

https://github.com/dotnet/runtime/blob/1e04f68b1b97bf36e4f1fc75ea23b9d156c19920/src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs#L1327-L1329

I could cache it in a `static readonly` field on this class, and just use `CultureInfo.InvariantCulture.TextInfo`. Would that resolve the concern?